Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -150,6 +151,93 @@ edge_length(typename boost::graph_traits<PolygonMesh>::edge_descriptor e,
   return edge_length(halfedge(e, pmesh), pmesh);
 }
 
+/**
+  * \ingroup measure_grp
+  * computes the squared length of an edge of a given polygon mesh.
+  * The edge is given by one of its halfedges, or the edge itself.
+  *
+  * @tparam PolygonMesh a model of `HalfedgeGraph`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param h one halfedge of the edge to compute the squared length
+  * @param pmesh the polygon mesh to which `h` belongs
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{vertex_point_map}
+  *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+  *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+  *                    as key type and `%Point_3` as value type}
+  *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+  *   \cgalParamNEnd
+  *
+ *   \cgalParamNBegin{geom_traits}
+ *     \cgalParamDescription{an instance of a geometric traits class}
+ *     \cgalParamType{a class model of `Kernel`}
+ *     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+ *     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+ *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  *
+  * @return the squared length of `h`. The return type `FT` is a number type. It is
+  * either deduced from the `geom_traits` \ref bgl_namedparameters "Named Parameters" if provided,
+  * or the geometric traits class deduced from the point property map
+  * of `pmesh`.
+  *
+  * @sa `edge_length()`
+  * @sa `face_border_length()`
+  */
+template<typename PolygonMesh,
+         typename NamedParameters>
+#ifdef DOXYGEN_RUNNING
+FT
+#else
+typename GetGeomTraits<PolygonMesh, NamedParameters>::type::FT
+#endif
+squared_edge_length(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor h,
+		    const PolygonMesh& pmesh,
+		    const NamedParameters& np)
+{
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+
```

Comment:
Could assert that h is non null

Reply:
Same question as for the preconditon.   @sloriot what is your opinion on this.   As @janetournois me just explaind there is an assertion for `face_area()`  and not for `edge_length()`  as it is "easier"  to have a null face, namely by writing `face_area(face(he, mesh), mesh)`  than having a null edge.

Reply:
I agree with Jane but it does not really hurt

Reply:
I added an assertion.